src/entity/Living.php

```diff
@@ -352,6 +353,19 @@ protected function onHitGround() : ?float{
 		return $newVerticalVelocity;
 	}
 
+	public function onInteract(Player $player, Vector3 $clickPos) : bool{
+		$item = $player->getInventory()->getItemInHand();
+		if($item->getTypeId() === ItemTypeIds::NAME_TAG && !$this instanceof Human && $item->hasCustomName()){
```

Comment:
`$this instanceof` defeats the purpose of OOP. Also, I'm not sure this logic belongs in `Living`.

Reply:
According to my tests and the wiki, all Living entities with the exception of EnderDragon and Human can be named with the nametag.

Reply:
> `$this instanceof` defeats the purpose of OOP

Solved.